sky/serve/replica_managers.py

```diff
@@ -67,10 +69,19 @@ def launch_cluster(task_yaml_path: str,
     """
     try:
         config = common_utils.read_yaml(os.path.expanduser(task_yaml_path))
+        resource_config = config.get('resources', {})
         if resources_override is not None:
-            resource_config = config.get('resources', {})
-            resource_config.update(resources_override)
+            if 'any_of' in resource_config:
+                for any_of_config in resource_config['any_of']:
+                    any_of_config.update(resources_override)
+            if 'ordered' in resource_config:
+                for any_of_config in resource_config['ordered']:
+                    any_of_config.update(resources_override)
+            else:
+                resource_config.update(resources_override)
```

Comment:
```suggestion
            resource_config.update(resources_override)
```
iiuc the outside will override the fields inside any_of and ordered?

Reply:
the any_of / ordered will override the outside fields. 
https://github.com/skypilot-org/skypilot/blob/master/sky/resources.py#L1189-L1194